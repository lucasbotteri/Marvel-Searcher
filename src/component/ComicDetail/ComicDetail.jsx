import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const ComicDetailWrapper = styled.div`
  align-items: start;
  column-gap: 20px;
  display: grid;
  grid-template-areas:
    "image title"
    "image creators"
    "image description"
    "image .";
  grid-template-columns: 50% auto;
  grid-template-rows: repeat(4, auto);
  justify-items: start;
  margin: auto;
  width: 66%;

  @media (max-width: 425) {
    width: 100%;
  }
`;

const ComicImage = styled.img`
  grid-area: image;
  justify-self: end;
  max-width: 100%;
  height: auto;
`;

const ComicTitle = styled.h2`
  font-size: ${(props) => props.theme.h2Size};
  grid-area: title;
`;

const ComicCreator = styled.p`
  font-size: 12px;
  font-weight: bold;
  margin: 0;
`;

const ComicCreatorsWrapper = styled.div`
  grid-area: creators;
`;

const ComicDescription = styled.p`
  font-size: 11px;
  grid-area: description;
`;

const ComicDetail = ({ comic }) => {
  return (
    <ComicDetailWrapper>
      <ComicImage
        src={comic.thumbnail}
        alt={`Comic ${comic.title} thumbnail`}
      />
      <ComicTitle>{comic.title}</ComicTitle>
      <ComicCreatorsWrapper>
        <ComicCreator>{`Published: ${comic.published}`}</ComicCreator>
        <ComicCreator>{`Writer: ${comic.writer}`}</ComicCreator>
        <ComicCreator>{`Penciler: ${comic.penciler}`}</ComicCreator>
        <ComicCreator>{`Cover Artist: ${comic.cover}`}</ComicCreator>
      </ComicCreatorsWrapper>
      <ComicDescription>{comic.description}</ComicDescription>
    </ComicDetailWrapper>
  );
};

ComicDetail.propTypes = {
  comic: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    thumbnail: PropTypes.string,
    description: PropTypes.string,
    writer: PropTypes.string,
    penciler: PropTypes.string,
    cover: PropTypes.string,
    published: PropTypes.string,
  }).isRequired,
};

export default ComicDetail;
