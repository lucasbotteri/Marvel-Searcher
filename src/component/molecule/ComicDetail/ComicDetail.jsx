import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const ComicDetailWrapper = styled.div`
  align-items: start;
  column-gap: 20px;
  display: grid;
  grid-template-areas:
    "image title"
    "image creators"
    "image description"
    "image .";
  grid-template-columns: 50% auto;
  grid-template-rows: repeat(4, auto);
  justify-items: start;
  margin: auto;
  width: 66%;

  @media (max-width: 425px) {
    grid-template-columns: 100%;
    grid-template-rows: repeat(5, auto);
    grid-template-areas:
      "image "
      "title"
      "creators"
      "description";
    justify-items: center;
    width: 90%;
  }
`;

const ComicImage = styled.img`
  grid-area: image;
  justify-self: end;
  max-width: 100%;
  height: auto;
  @media (max-width: 425px) {
    justify-self: center;
  }
`;

const ComicTitle = styled.h2`
  font-size: ${(props) => props.theme.h2Size};
  grid-area: title;
  text-align: start;

  @media (max-width: 425px) {
    text-align: center;
  }
`;

const ComicCreator = styled.p`
  font-size: 12px;
  font-weight: bold;
  margin: 0;
`;

const ComicCreatorsWrapper = styled.div`
  grid-area: creators;
  text-align: start;

  @media (max-width: 425px) {
    text-align: center;
  }
`;

const ComicDescription = styled.p`
  font-size: 11px;
  grid-area: description;
  text-align: start;

  @media (max-width: 425px) {
    text-align: center;
  }
`;

const ComicDetail = ({ comic }) => {
  return (
    <ComicDetailWrapper>
      <ComicImage
        src={comic.thumbnail}
        alt={`Comic ${comic.title} thumbnail`}
      />
      <ComicTitle>{comic.title}</ComicTitle>
      <ComicCreatorsWrapper>
        <ComicCreator>{`Published: ${comic.published}`}</ComicCreator>
        <ComicCreator>{`Writer: ${comic.writer}`}</ComicCreator>
        <ComicCreator>{`Penciler: ${comic.penciler}`}</ComicCreator>
        <ComicCreator>{`Cover Artist: ${comic.cover}`}</ComicCreator>
      </ComicCreatorsWrapper>
      <ComicDescription>{comic.description}</ComicDescription>
    </ComicDetailWrapper>
  );
};

ComicDetail.propTypes = {
  comic: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    thumbnail: PropTypes.string,
    description: PropTypes.string,
    writer: PropTypes.string,
    penciler: PropTypes.string,
    cover: PropTypes.string,
    published: PropTypes.string,
  }).isRequired,
};

export default ComicDetail;
