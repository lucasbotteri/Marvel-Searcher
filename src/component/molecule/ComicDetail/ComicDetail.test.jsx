import React from "react";
import { shallow } from "enzyme";
import ComicDetail from "./ComicDetail";

const comic = {
  id: 70718,
  title: "The Amazing Spider-Man (2018) #22",
  published: "2019-05-29T00:00:00-0400",
  thumbnail:
    "http://i.annihil.us/u/prod/marvel/i/mg/2/d0/5ce401142935b/portrait_uncanny.jpg",
  description:
    "HUNTED AFTERMATH! The fallout from “HUNTED” continues to loom, and much of Peter’s life is called into question. What is left of Spider-Man after living through the harrowing hunt?! ",
  writer: "Nick Spencer",
  penciler: "Unknown",
  cover: "Unknown",
};

describe("ComicDetail", () => {
  it("Should render without crashing", () => {
    const wrapper = shallow(<ComicDetail comic={comic} />);
    expect(wrapper).toBeTruthy();
  });

  it("Should match snapshot", () => {
    const wrapper = shallow(<ComicDetail comic={comic} />);
    expect(wrapper).toMatchSnapshot();
  });
});
