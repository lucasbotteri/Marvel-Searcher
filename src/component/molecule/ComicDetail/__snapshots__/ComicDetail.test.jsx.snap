// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard Should match snapshot 1`] = `ShallowWrapper {}`;

exports[`ComicDetail Should match snapshot 1`] = `ShallowWrapper {}`;
